### PR TITLE
Mark min-svn fields as deprecated.

### PR DESCRIPTION
**Pull request by bluegate010**

### CODE REVIEW

File: image/verify/src/verifier.rs [hunk 1]
```diff
@@ -714,7 +714,8 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
                 Err(CaliptraError::IMAGE_VERIFIER_ERR_FMC_SVN_GREATER_THAN_MAX_SUPPORTED)?;
             }
 
-            if verify_info.svn < verify_info.min_svn {
+            // TODO: remove this check.
+            if verify_info.svn < verify_info.deprecated_min_svn {
```
Comment (korran) [inline image/verify/src/verifier.rs:718]: Aha, so we are looking at this field...

Comment (bluegate010) [reply image/verify/src/verifier.rs:718]: Yes but it's just a sanity check, comparing two numbers in the signed header against each other. No other logic being performed on the field. Would be a safe two-liner change to just remove this field.

**Review by FerralCoder — APPROVED** [review]

File: builder/src/lib.rs [hunk 2]
```diff
@@ -408,10 +408,10 @@ pub fn elf_size(elf_bytes: &[u8]) -> io::Result<u64> {
 #[derive(Clone)]
 pub struct ImageOptions {
     pub fmc_version: u32,
-    pub fmc_min_svn: u32,
+    pub deprecated_fmc_min_svn: u32,
```
Comment (mhatrevi) [inline builder/src/lib.rs:411]: Can we name this 'reserved'?

File: drivers/src/fuse_log.rs [hunk 3]
```diff
@@ -19,16 +19,16 @@ use zeroize::Zeroize;
...
-    FuseFmcSvn = 5,                 // 4 bytes
-    ManifestRtSvn = 6,              // 4 bytes
-    ManifestRtMinSvn = 7,           // 4 bytes
-    FuseRtSvn = 8,                  // 4 bytes
-    VendorLmsPubKeyIndex = 9,       // 4 bytes  (From Manifest)
-    VendorLmsPubKeyRevocation = 10, // 4 bytes  (From Fuse)
+    VendorEccPubKeyIndex = 1,        // 4 bytes  (From Manifest)
+    VendorEccPubKeyRevocation = 2,   // 4 bytes  (From Fuse)
+    ManifestFmcSvn = 3,              // 4 bytes
+    ManifestDeprecatedFmcMinSvn = 4, // 4 bytes
```
Comment (mhatrevi) [inline drivers/src/fuse_log.rs:25]: Can we name this to 'reserved'?

File: image/elf/src/lib.rs [hunk 4]
```diff
@@ -24,7 +24,7 @@ use std::path::PathBuf;
 pub struct ElfExecutable {
     version: u32,
     svn: u32,
-    min_svn: u32,
+    deprecated_min_svn: u32,
```
Comment (mhatrevi) [inline image/elf/src/lib.rs:27]: This can be removed, right?

File: image/elf/src/lib.rs [hunk 5]
```diff
@@ -140,9 +140,9 @@ impl ImageGenratorExecutable for ElfExecutable {
         self.svn
     }
 
-    /// Executable Minimum Security Version Number
-    fn min_svn(&self) -> u32 {
-        self.min_svn
+    /// Executable deprecated minimum SVN
+    fn deprecated_min_svn(&self) -> u32 {
```
Comment (mhatrevi) [inline image/elf/src/lib.rs:144]: We can remove this.

File: image/types/src/lib.rs [hunk 6]
```diff
@@ -372,8 +372,8 @@ pub struct ImageTocEntry {
     /// Security Version Number
     pub svn: u32,
 
-    /// Minimum Security Version Number
-    pub min_svn: u32,
+    /// Deprecated field, must be less than or equal to SVN.
+    pub deprecated_min_svn: u32,
```
Comment (mhatrevi) [inline image/types/src/lib.rs:376]: Can we name this to 'reserved'?

File: image/verify/src/verifier.rs (hunk 1 above)
Comment (mhatrevi) [reply image/verify/src/verifier.rs:718]: Can we remove this?

File: image/verify/src/verifier.rs [hunk 7]
```diff
@@ -749,7 +750,7 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
 
         let log_info: ImageSvnLogInfo = ImageSvnLogInfo {
             manifest_svn: verify_info.svn,
-            manifest_min_svn: verify_info.min_svn,
+            manifest_deprecated_min_svn: verify_info.deprecated_min_svn,
```
Comment (mhatrevi) [inline image/verify/src/verifier.rs:753]: We should just rename the field to 'reserved' in the image and remove it from verifier function, test cases etc.

**Review by mhatrevi — CHANGES_REQUESTED** [review]
IMO, we should rename the field to 'reserved' in the image and remove it from other places (verifier, tests etc)

File: image/verify/src/verifier.rs (hunk 1 above)
Comment (bluegate010) [reply image/verify/src/verifier.rs]: Will do in a follow-up.

File: builder/src/lib.rs (hunk 2 above)
Comment (bluegate010) [reply builder/src/lib.rs]: Have removed.

**Comment by bluegate010** [comment]
> IMO, we should rename the field to 'reserved' in the image and remove it from other places (verifier, tests etc)

Have removed test cases and builder code related to this, marked the field as reserved in the firmware header and fuse log record, and will issue a follow-up PR to remove the remaining logic in ROM that operates on this field.

File: image/verify/src/verifier.rs [hunk 8]
```diff
@@ -749,7 +750,7 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
 
         let log_info: ImageSvnLogInfo = ImageSvnLogInfo {
             manifest_svn: verify_info.svn,
-            manifest_min_svn: verify_info.min_svn,
+            reserved: 0,
```
Comment (korran) [inline image/verify/src/verifier.rs:753]: This is probably causing your ROM diff...

File: image/verify/src/verifier.rs [hunk 9]
```diff
@@ -827,7 +829,7 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
...
```
Comment (korran) [inline image/verify/src/verifier.rs:832]: and here

File: image/verify/src/verifier.rs (hunk 8 above)
Comment (bluegate010) [reply image/verify/src/verifier.rs]: Ah, fixed.

**Review by mhatrevi — APPROVED** [review]

**Review by rusty1968 — APPROVED** [review]

**Review by korran — APPROVED** [review]

**Pull request merged**